Source/WebSocket.swift

```diff
@@ -1668,13 +1668,17 @@ public class WebSocket: NSObject {
     public init(request: NSURLRequest, subProtocols : [String] = []){
         opened = true
         ws = InnerWebSocket(request: request, subProtocols: subProtocols, stub: false)
+        super.init()
+        ws.eclose = { [unowned self] in
```

Comment:
What do you think an `internalInit` or something similiar to avoid duplicated code?

Reply:
not working to me, socket opened again but does not got any event anymore